### PR TITLE
HostAliases support for Scheduler, Worker, Trigger and Webserver

### PR DESCRIPTION
**Pull request by rrcrrcrrc**
This PR adds `scheduler.hostAliases` and `webserver.hostAliases` parameter in helm chart in order to use [kubernetes HostAliases](https://kubernetes.io/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases/) in scheduler and webserver pods.

### CODE REVIEW

**Comment by rrcrrcrrc** [comment]
Can we add this feature to Helm Chart 1.7.0?

**Comment by potiuk** [comment]
> Can we add this feature to Helm Chart 1.7.0?

First it would have to be reviewed and apprved as all other PRs. You need to exercise a bit of patience to get there.

File: tests/charts/test_scheduler.py [hunk 1]
```diff
@@ -115,6 +115,20 @@ def test_should_add_extra_volume_and_extra_volume_mount(self):
             "spec.template.spec.containers[0].volumeMounts[*].name", docs[0]
         )
 
+    def test_scheduler_host_aliases(self):
+        docs = render_chart(
+            values={
+                "executor": "CeleryExecutor",
```
Comment (jedcunningham) [inline tests/charts/test_scheduler.py]:
```suggestion
```

Don't need that, it's the default.

File: chart/values.yaml [hunk 2]
```diff
@@ -685,6 +685,16 @@ scheduler:
   #      weight: 100
   tolerations: []
   topologySpreadConstraints: []
+  # hostAliases to use in scheduler pods.
+  # See:
+  # https://kubernetes.io/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases/
```
Comment (jedcunningham) [inline chart/values.yaml]: I think we can get away without these.

```suggestion
```

File: chart/values.yaml [hunk 3]
```diff
@@ -685,6 +685,16 @@ scheduler:
...
+  hostAliases: []
+  # - ip: "127.0.0.2"
+  #   hostnames:
+  #   - "test.hostname.one"
+  # - ip: "127.0.0.3"
+  #   hostnames:
+  #   - "test.hostname.two"
```
Comment (jedcunningham) [inline chart/values.yaml]:
```suggestion
```

I think one example should be sufficient?

File: chart/values.schema.json [hunk 4]
```diff
@@ -1802,6 +1802,28 @@
...
+                            "ip": "127.0.0.2",
+                            "hostnames": [
+                                "test.hostname.one"
+                            ]
+                        },
+                        {
+                            "ip": "127.0.0.3",
+                            "hostnames": [
+                                "test.hostname.two"
+                            ]
```
Comment (jedcunningham) [inline chart/values.schema.json]:
```suggestion
```

Here as well.

**Review by jedcunningham — COMMENTED** [review]
We should also do this for the other Airflow components (workers, triggerer, etc). Up to you if you want to do it here or leave it for someone else to implement.

**Comment by rrcrrcrrc** [comment]
Already edited what you comment and added hostAlises to the triggerer component too. This would make hostAliases available for Scheduler, Worker, Triggerer and Webserver.

File: chart/templates/triggerer/triggerer-deployment.yaml [hunk 5]
```diff
@@ -99,7 +99,11 @@ spec:
       tolerations:
         {{- toYaml $tolerations | nindent 8 }}
       topologySpreadConstraints:
-        {{- toYaml $topologySpreadConstraints | nindent 8 }}
+        {{- toYaml $topologySpreadConstraints | nindent 8 }}
```
Comment (jedcunningham) [inline chart/templates/triggerer/triggerer-deployment.yaml]:
```suggestion
        {{- toYaml $topologySpreadConstraints | nindent 8 }}
```

**Review by jedcunningham — APPROVED** [review]

File: chart/values.schema.json [hunk 6]
```diff
@@ -2148,6 +2158,22 @@
                         "$ref": "#/definitions/io.k8s.api.core.v1.TopologySpreadConstraint"
                     }
                 },
+                "hostAliases": {
+                    "description": "Specify HostAliases for trigerer.",
```
Comment (jedcunningham) [inline chart/values.schema.json]:
```suggestion
                    "description": "Specify HostAliases for triggerer.",
```

File: tests/charts/test_webserver.py [hunk 7]
```diff
@@ -216,6 +216,19 @@ def test_should_add_extra_init_containers(self):
...
+                "webserver": {
+                    "hostAliases": [{"ip": "127.0.0.2", "hostnames": ["test.hostname"]}],
+                },
+            },
+            show_only=["templates/webserver/webserver-deployment.yaml"],
+        )
+
+        assert "127.0.0.2" == jmespath.search("spec.template.spec.hostAliases[0].ip", docs[0])
+        assert "test.hostname" == jmespath.search("spec.template.spec.hostAliases[0].hostnames[0]", docs[0])
+
```
Comment (jedcunningham) [inline tests/charts/test_webserver.py]:
```suggestion

```

File: tests/charts/test_webserver.py [hunk 8]
```diff
@@ -216,6 +216,18 @@ def test_should_add_extra_init_containers(self):
...
+            values={
+                "webserver": {
+                    "hostAliases": [{"ip": "127.0.0.2", "hostnames": ["test.hostname"]}],
+                },
+            },
+            show_only=["templates/webserver/webserver-deployment.yaml"],
+        )
+
+        assert "127.0.0.2" == jmespath.search("spec.template.spec.hostAliases[0].ip", docs[0])
+        assert "test.hostname" == jmespath.search("spec.template.spec.hostAliases[0].hostnames[0]", docs[0])
```
Comment (jedcunningham) [inline tests/charts/test_webserver.py:230]:
```suggestion
        assert "test.hostname" == jmespath.search("spec.template.spec.hostAliases[0].hostnames[0]", docs[0])

```

**Review by pingzh — APPROVED** [review]

**Review by jedcunningham — CHANGES_REQUESTED** [review]
Looks like you accidentally deleted some non-helm tests?

**Comment by rrcrrcrrc** [comment]
> Looks like you accidentally deleted some non-helm tests?

Yes @jedcunningham , my fault, just wanna change the "indent" to "nindent". Can someone do it for all? This week I got not PC

**Comment by rrcrrcrrc** [comment]
Hi, I don´t know what to do this this PR rigth now.. Literally is my first PR

**Comment by potiuk** [comment]
> Hi, I don´t know what to do this this PR rigth now.. Literally is my first PR

If you deleted stuff accidentally, If you have no idea where things get wrong - close it, open a new one and start from scratch copying new code. This is nothing Airflow specific (though we use rebase workflow) - but you can read CONTRIBUTING.rst and possibly search the internet on rebasing, solving conflicts etc. If you have no experience, then start small and take your time to learn. No hurry - you can even take online courses. When you want to contribute to open-source you need ot be self-driven , self-learning and self sufficient and while some help and guidance is given, usually it is in the form of comments in PR for concrete changes or documentation (like CONTRIBUTING) written down for you to read and learn rathe than individual hand-holding and instructing step-by-step

**Pull request closed without merging**